server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -443,8 +436,12 @@ public synchronized void open(String address) throws IOException {
       byte[] cryptoParams = encrypter.getDecryptionParameters();
       CryptoUtils.writeParams(cryptoParams, logFile);
 
-      encryptingLogFile = new DataOutputStream(
-          encrypter.encryptStream(new NoFlushOutputStream(logFile)));
+      // write directly to logFile if no crypto detected
+      if (encrypter instanceof NoFileEncrypter) {
+        encryptingLogFile = logFile;
+      } else {
+        encryptingLogFile = new NoFlushOutputStream(encrypter.encryptStream(logFile));
```

Comment:
@PircDef do you have any thoughts on this?

Reply:
I think `encryptingLogFile` could be of type `DataOutput`.  Based on this we could change the if to the following.   This will accomplish the same thing with less assumptions. 

```java
OutputStream tmp = encrypter.encryptStream(logFile);
if(tmp instanceof DataOutput) {
  encryptingLogFile = (DataOutput) tmp;
} else {
  encryptingLogFile = NoFlushOutputStream(encrypter.encryptStream(logFile));
}
```

Reply:
If we need the type to be a `DataOutput`, then wouldn't it be better for `encrypter.encryptStream(logFile)` to just return `DataOutputStream`, and let the implementation decide whether to wrap it with `NoFlushOutputStream` or not?

Reply:
I am starting to think the entire no flush thing is suspicious in that it may report back that things are are flushed to a batch writer when in fact they are not.

Reply:
I was thinking this as well but we don't want the NoFlushoutputStream for RFile encryption.  I am testing a change where AESCryptoService returns the NoFlushOutputStream so that detail is transparent to the DfsLogger